api/models/v0.3/modelPolicy.js

```diff
@@ -1,140 +1,134 @@
-  ptype() {
-    return this.policy["ptype"];
-  }
-  // Return the card expiration hash associated with the given patron policy
-  card_type() {
-    return this.policy["card_type"];
-  }
-  required_fields() {
-    return this.policy["required_fields"];
-  }
-  minimum_age() {
-    return this.policy["minimum_age"];
-  }
-  service_area() {
-    return this.policy["service_area"];
-  }
-  return_policy_field(field) {
-    return this.policy[field];
-  }
-  is_default() {
-    return this.policy_type === DEFAULT_POLICY_TYPE;
-  }
-  is_required(field) {
-    return this.required_fields().includes(field);
-  }
-  determine_ptype(patron) {
-    if (this.service_area() && this.ptype.has_key("metro")) {
-      if (patron.lives_or_works_in_city) {
-        return this.ptype["metro"]["id"];
-      } else if (patron.lives_in_state) {
-        return this.ptype["default"]["id"];
+  const policyField = (field) => policy[field];
+  const isRequiredField = (field) => policyField('requiredFields').includes(field);
+  const determinePtype = (patron = undefined) => {
+    const ptype = policyField('ptype');
+    const hasServiceArea = policyField('serviceArea')
+      && Object.keys(policyField('serviceArea')).length;
+    const hasMetroKey = Object.keys(ptype).includes('metro');
+    if (hasServiceArea && hasMetroKey) {
+      if (patron.livesOrWorksInCity()) {
+        return ptype.metro.id;
+      }
+      if (patron.livesInState()) {
+        return ptype.default.id;
       }
     }
-    return this.ptype["default"]["id"];
-  }
-  is_web_applicant() {
-    return this.policy_type === "web_applicant";
-  }
-  determine_agency(patron_params) {
-    if (is_web_applicant) {
+    return ptype.default.id;
+  };
+  const determineAgency = (patronParams = {}) => {
+    if (isWebApplicant) {
       if (
-        patron_params["patron_agency"] &&
-        parseInt(patron_params["patron_agency"], 10) === 199
+        patronParams
+        && patronParams.patronAgency
+        && parseInt(patronParams.patronAgency, 10) === 199
       ) {
-        this.policy["agency"] = IlsHelper.WEB_APPLICANT_NYS_AGENCY;
+        policy.agency = IlsHelper.WEB_APPLICANT_NYS_AGENCY;
       } else {
-        this.policy["agency"] = IlsHelper.WEB_APPLICANT_AGENCY;
+        policy.agency = IlsHelper.WEB_APPLICANT_AGENCY;
       }
     } else {
-      this.policy["agency"] = IlsHelper.DEFAULT_PATRON_AGENCY;
+      policy.agency = IlsHelper.DEFAULT_PATRON_AGENCY;
     }
-  }
+
+    return policy.agency;
+  };
 
   // Validations
-  uses_an_approved_policy() {
-    const keys = Object.keys(Policy.ils_policy);
-    if (!keys.include(this.policy_type)) {
-      this.errors.push(`${policy_type} is invalid, must be ${keys.join(", ")}`);
+  const usesAnApprovedPolicy = () => {
+    const keys = Object.keys(ilsPolicy);
+    if (!keys.includes(policyType)) {
+      console.log(
+        `${policyType} policy type is invalid, must be of type ${keys.join(
+          ', ',
```

Comment:
Why the extra comma?

Reply:
It's not extra, it's used to join the keys in the `ilsPolicy` object.

Reply:
The first comma (the one in the single quotes, with the space after it) is joining the keys.  But then there's a second one...?

Reply:
Ah, I see. It's `eslint`'s dangling comma rule. Usually, it's done in objects and it is a bit weird seeing it here.